code/modules/mob/living/carbon/xenomorph/castes/king/abilities_king.dm

```diff
@@ -32,104 +32,272 @@
+	if(!do_after(owner, PETRIFY_WINDUP_TIME, FALSE, owner, BUSY_ICON_DANGER))
+		flick("eye_closing", eye)
+		addtimer(CALLBACK(src, .proc/remove_eye, eye), 7, TIMER_CLIENT_TIME)
+		return
+	playsound(owner, 'sound/effects/petrify_activate.ogg', 50)
+	var/list/mob/living/carbon/human/humans = list()
+	for(var/mob/living/carbon/human/human in view(PETRIFY_RANGE, owner))
+		if(human.stat != CONSCIOUS)
+			continue
+		if(is_blind(human))
+			continue
+		humans += human
+		human.notransform = TRUE
+		human.status_flags |= GODMODE
+		human.add_atom_colour(COLOR_GRAY, TEMPORARY_COLOUR_PRIORITY)
+		human.log_message("has been petrified by [owner] for [PETRIFY_DURATION] ticks", LOG_ATTACK, color="pink")
+		var/image/stone_overlay = image('icons/effects/effects.dmi', null, "petrified_overlay")
+		stone_overlay.filters += filter(arglist(alpha_mask_filter(render_source="*[REF(human)]",flags=MASK_INVERSE)))
+		var/mutable_appearance/mask = mutable_appearance()
+		mask.appearance = human.appearance
+		mask.render_target = "*[REF(human)]"
+		mask.alpha = 125
+		stone_overlay.overlays += mask
+		human.overlays += stone_overlay
+		humans[human] = stone_overlay
+
+	if(!length(humans))
+		flick("eye_closing", eye)
+		addtimer(CALLBACK(src, .proc/remove_eye, eye), 7, TIMER_CLIENT_TIME)
+		return
+	addtimer(CALLBACK(src, .proc/remove_eye, eye), 10, TIMER_CLIENT_TIME)
+	flick("eye_explode", eye)
+	addtimer(CALLBACK(src, .proc/end_effects, humans), PETRIFY_DURATION)
+	add_cooldown()
+	succeed_activate()
+
+///ends all combat-relazted effects
+/datum/action/xeno_action/petrify/proc/end_effects(list/humans)
+	for(var/mob/living/carbon/human/human AS in humans)
+		human.notransform = FALSE
+		human.status_flags &= ~GODMODE
+		human.remove_atom_colour(TEMPORARY_COLOUR_PRIORITY, COLOR_GRAY)
+		human.overlays -= humans[human]
+
+///callback for removing the eye from viscontents
+/datum/action/xeno_action/petrify/proc/remove_eye(obj/effect/eye)
+	owner.vis_contents -= eye
+
+// ***************************************
+// *********** Off-Guard
+// ***************************************
+/datum/action/xeno_action/activable/off_guard
+	name = "Off-guard"
+	action_icon_state = "off_guard"
+	mechanics_text = "Muddles the mind of an enemy, making it harder for them to focus their aim for a while."
+	ability_name = "off guard"
+	plasma_cost = 100
+	cooldown_timer = 20 SECONDS
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
+	target_flags = XABB_MOB_TARGET
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_OFFGUARD,
+	)
+
+
+/datum/action/xeno_action/activable/off_guard/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return
+	if(!ishuman(A))
+		if(!silent)
+			A.balloon_alert(owner, "not human")
+		return FALSE
+	var/mob/living/carbon/human/target = A
+	if(target.stat == DEAD)
+		if(!silent)
+			target.balloon_alert(owner, "already dead")
+		return FALSE
+
+/datum/action/xeno_action/activable/off_guard/use_ability(atom/target)
```

Comment:
You can't press the hotkey to select the ability, which is a fairly useful QOL thing. See warlock psyblast.